internal/operator-controller/contentmanager/cache/cache_test.go

```diff
@@ -69,11 +70,11 @@ func TestCacheWatch(t *testing.T) {
 	)
 
 	pod := &corev1.Pod{}
-	podGvk := corev1.SchemeGroupVersion.WithKind("Pod")
-	pod.SetGroupVersionKind(podGvk)
+	pod.SetGroupVersionKind(corev1.SchemeGroupVersion.WithKind("Pod"))
+	pod.SetNamespace(rand.String(8))
 
 	require.NoError(t, c.Watch(context.Background(), &mockWatcher{}, pod))
-	require.Contains(t, c.(*cache).sources, podGvk, "sources", c.(*cache).sources)
+	require.Contains(t, c.(*cache).sources, sourceKey{pod.Namespace, pod.GroupVersionKind()}, "sources", c.(*cache).sources)
```

Comment:
Probably would require a bit of test refactoring (or a new test function), but I would recommend adding a test case to cover a cluster scoped resource being provided with a namespace set to ensure you are appropriately creating a cluster scoped source instead of a namespace scoped one.

Reply:
Added a new test!